src/components/forms/MemoryLimitAvailable.tsx

```diff
@@ -1,59 +1,134 @@
+import type { CreateInstanceFormValues } from "pages/instances/CreateInstance";
+import type { EditInstanceFormValues } from "pages/instances/EditInstance";
+import { CLUSTER_GROUP_PREFIX } from "util/instances";
+import ResourceLink from "components/ResourceLink";
+import { useIsClustered } from "context/useIsClustered";
+import { useResources } from "context/useResources";
 
+interface MemoryLimit {
+  min: number;
+  max: number;
+  source: "project" | "cluster-member" | "";
+}
 interface Props {
+  formik?: InstanceAndProfileFormikProps;
   project?: LxdProject;
 }
 
-const MemoryLimitAvailable: FC<Props> = ({ project }) => {
+const MemoryLimitAvailable: FC<Props> = ({ formik, project }) => {
   const notify = useNotify();
-  const { canViewResources } = useServerEntitlements();
-
-  const {
-    data: resources,
-    error,
-    isLoading,
-  } = useQuery({
-    queryKey: [queryKeys.resources],
-    queryFn: async () => fetchResources(),
-    enabled: canViewResources(),
-  });
+  const isClustered = useIsClustered();
+  const values = formik?.values;
+  const projectResourceMemory = limitToBytes(project?.config["limits.memory"]);
+  const { target } = values as CreateInstanceFormValues;
+  const { location } = values as EditInstanceFormValues;
+  const isClusterGroupTarget = target?.startsWith(CLUSTER_GROUP_PREFIX);
+  const validTarget = target && !isClusterGroupTarget ? target : undefined;
+  const clusterMember = isClustered ? (validTarget ?? location) : undefined;
+
+  const { data: resources, error, isLoading } = useResources(clusterMember);
 
   if (isLoading) {
     return <Spinner className="u-loader" text="Loading resources..." />;
   }
-
   if (error) {
     notify.failure("Loading resources failed", error);
+    return null;
   }
+  if (!resources) {
+    return null;
+  }
+
+  const getSourceQualifier = (source: string) => {
+    if (source === "project") {
+      return (
+        <>
+          {"in project "}
+          <ResourceLink
+            type="project"
+            value={project?.name || "default"}
+            to={`/ui/projects/${encodeURIComponent(project?.name || "default")}`}
+          />
+        </>
+      );
+    }
 
-  const getAvailableMemory = () => {
-    if (!project?.config["limits.memory"]) {
-      return resources?.memory.total;
+    if (source === "cluster-member") {
+      return (
+        <>
+          {"on cluster member "}
+          <ResourceLink
+            type="cluster-member"
+            value={clusterMember ?? ""}
+            to={`/ui/cluster/member/${encodeURIComponent(clusterMember ?? "")}`}
+          />
+        </>
+      );
     }
-    if (!resources?.memory.total) {
-      return limitToBytes(project.config["limits.memory"]);
+    return null;
+  };
+
+  const getAvailableMemory = (): MemoryLimit => {
+    const resourceArray = Array.isArray(resources) ? resources : [resources];
+    const projectLimit = Number(projectResourceMemory);
+
+    if (resourceArray.length === 0) {
```

Comment:
```suggestion
    if (resourceArray.length === 0 || projectLimit < minResourceMemory ) {
```
If the project limit is the lower bound, we should report it as such.